feat: add documentation on placeholder dynamic documentation (#275)
AmadeusITGroup/otter: docs/rules-engine/how-to-use/placeholders.md

This behavior is based on the fact that a real user rarely goes back and forth with the language update.

### Dynamic localization

> If your placeholder template HTML does not differ between languages, you might want to consider a dynamic localization.

Reviewer: ```suggestion
If your HTML template placeholder remains the same across different languages, you might want to consider a dynamic localization.
```

> If your HTML template placeholder remains the same across different languages, you might want to consider a dynamic localization.

Reviewer 2: Not sure this documentation is really linked to the rules engine or for the global usage of the placeholder mechanism.
Maybe it should be part of the placeholder documentation or at least be referred in it

Author: Well this documentation already contains many things on how to setup your placeholder and your PR is not yet merged.
This is a deadlock 😆 
Either we first merge this PR and you integrate this in yours or I will do it in this pr

#### Implementation
You can activate the dynamic localization feature in your placeholder by following this example:
```json
{
  "vars": {
    "keyToTranslation": {
      "type": "localisation",
      "value": "my-localisation-key"
    }
  },
  "template": "<div class=\"fancy-box\"><%= keyToTranslation %></div>"
}
```
Your key should be described in a placeholder variable where the type will be ``localisation`` and its value 
the translation key itself. You can refer to the variable in your template thanks to the variable key (``keyToTranslation`` 
in the example above).

**Note**: you should **not** create [LANGUAGE] files for placeholders that use the dynamic localization feature. The static
localization is not compatible with the dynamic localization.

#### ICU and parameter support

Today, the Otter Framework supports the ICU syntax as well as parameters. You only need to bind your parameter with a fact
sharing the same name.

As an example, let's create a counter that will emit every second.
You will first need to design your placeholder referencing the translation key and the facts it is bound to.

> Let's consider what this placeholder would look like if it were completely integrated in your angular component

Author: If it were -> let's work on an hypothesis

```html
"<div style=\"border-radius:10%; background:red;\">{{'o3r-increment-key' | translate}}</div>"
```

Then, let's create a new localization key for each of your supported language:

* en-GB.json
```json
{
  "o3r-increment-key": "{increment, plural, =1 {1 second has} other {{{increment}} seconds have}} elapsed since you opened the page"
}
```
* fr-FR.json
```json
{
  "o3r-increment-key": "Cela fait {increment, plural, =1 {1 seconde}} other {{{increment} secondes} que tu as ouvert cette page"
}
```

Note that the ``o3r-increment-key`` translations take ``increment`` as a parameter. This means you need to create an ``increment`` 
fact to fill the value. You can follow the [fact creation documentation](./create-custom-fact.md).

> ```

Reviewer: ```suggestion
```typescript
```

import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { FactsService, RulesEngineService } from '@o3r/rules-engine';
import { interval } from 'rxjs';
import { retrieveUrl } from './fact-factories/index';
import { PageFacts } from './page.facts';

@Injectable()
export class PageFactsService extends FactsService<PageFacts> {

  public facts = {
    pageUrl: this.router.events.pipe(retrieveUrl()),
    // This is for demo app testing only, don't do this in a real application

>     increment: interval(2000)

Reviewer 3: Maybe put `1000` to have an interval for each second

Author: it spam slightly less

}
```

**Limitations:**
This feature deeply binds functional facts exposed in your application to your translations. 
You will need to carefully plan the way you bind your localization key to your facts to avoid messy references.

For example, let's imagine you want a generic counters until specific events. 
You will probably want to reuse your placeholder in different pages for different events:

```json
{
  "vars": {
    "titleKey": {
      "type": "localisation",
      "value": "o3r-event-title-key"
    },
    "contentWithCounterKey": {
      "type": "localisation",
      "value": "o3r-event-counter-key",
      "vars": ["increment"]
    },
    "increment": {
      "type": "fact",
      "value": "increment"
    }
  },
  "template": "<div style=\"fancy-style: with-many-properties;\"><span style=\"fancy-title-style: nice-properties;\"><%= titleKey %></span><div><%= contentWithCounterKey %></div></div>"
}
```

You might be tempted to use this generic template for all your events but the value of your counter parameter will depend 
on the event itself (Easter, next Summer Holidays for example). 

> This means that ``increment`` might have different value depending on the context of the page which might be tricky to maintain.  

Reviewer 4: a different value OR different values

> You should also be careful and consider the cost of multiplying keys to fit each of your use case as redundancy can decrease

Reviewer 4: use cases
